src/org/linphone/setup/SetupActivity.java

```diff
@@ -69,17 +73,20 @@
 	final String registrarSRVLookupFormatTLS="_sips._tcp.%domain%";
 	//URL format for autoConfig lookup
 	final String autoConfigSRVLookupFormat="_rueconfig._tls.%domain%";
+	private Handler mHandler;
+	private ServiceWaitThread mThread;
 
 	static int WIFI_ACTIVITY_RESULT=0;
-
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
 //		if (getResources().getBoolean(R.bool.isTablet) && getRequestedOrientation() != ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
 //        	setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
 //        }
+		mHandler = new Handler();
 		
 		setContentView(R.layout.setup);
+		CDNProviders.getInstance().setContext(getApplicationContext());
 		firstFragment = getResources().getBoolean(R.bool.setup_use_linphone_as_first_fragment) ?
```

Comment:
On Nexus 6 I'm experiencing a crash on login from a null pointer exception in CDNProviders line:127. It looks like sharedPreferences is not being initialized when trying to access it's editor. Ensure you are performing sanity checks and that getApplicationContext() does not return null. You may need to use setContext(SetupActivity.this);